src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,126 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final int GROWTH_FACTOR = 2;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int capacity;
+    private int threshold;
+    private Node<K, V>[] buckets;
+    private int size;
+
+    public MyHashMap() {
+        this.buckets = new Node[INITIAL_CAPACITY];
+        this.capacity = INITIAL_CAPACITY;
+        this.threshold = (int) (INITIAL_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
+            resize();
+        }
+        if (key == null) {
+            putForNullKey(value);
+            return;
+        }
+        int index = getIndexOfNodeByKey(key);
+        Node<K, V> node = buckets[index];
+        while (node != null) {
+            if (keyEquals(node.key, key)) {
+                node.value = value;
+                return;
+            }
+            node = node.next;
+        }
+        addNode(key, value, index);
     }
 
     @Override
     public V getValue(K key) {
+        if (key == null) {
+            return getValueForNullKey();
+        }
+        int index = getIndexOfNodeByKey(key);
+        Node<K, V> node = buckets[index];
+        while (node != null) {
+            if (keyEquals(node.key, key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void putForNullKey(V value) {
+        Node<K, V> node = buckets[0];
+        while (node != null) {
+            if (node.key == null) {
+                node.value = value;
+                return;
+            }
+            node = node.next;
+        }
+        addNode(null, value, 0);
+    }
+
+    private void addNode(K key, V value, int index) {
```

Comment:
This method should create new node and add this new after last one in "linked list"